### DISCUSSION TITLE
Can I install the arm64_ventura package bottle on a computer with apple silicon cpu and sequoia macos

### DISCUSSION THREAD

**Discussion by RealChuan** [Homebrew/discussions#5721] [Getting started] [answered]
### Output of `brew config`

```shell
-
```

### Output of `brew doctor`

```shell
-
```

### Description of issue

-

**Comment by SMillerDev** [comment, marked as answer]
Homebrew will pick the latest available bottle.

**Reply by RealChuan** [reply]
Can I manually specify the bottle version

**Reply by SMillerDev** [reply]
`brew fetch` exists to manually fetch bottles, but I'm not certain you can install those.

**Reply by RealChuan** [reply]
mpv-0.39.0_1.arm64_sequoia.bottle.tar.gz 27-Oct-2024 02:11 4167841
mpv-0.39.0_1.arm64_sonoma.bottle.tar.gz 27-Oct-2024 02:11 4167680
mpv-0.39.0_1.arm64_ventura.bottle.tar.gz 27-Oct-2024 02:11 4197167
mpv-0.39.0_1.sonoma.bottle.tar.gz 27-Oct-2024 02:11 4228357
mpv-0.39.0_1.ventura.bottle.tar.gz 27-Oct-2024 02:12 4253494
mpv-0.39.0_1.x86_64_linux.bottle.tar.gz 27-Oct-2024 02:12 3052961

The packages on bottles have different architecture fields. Can I modify some code to make the download url is arm64_ventura?

**Reply by SMillerDev** [reply]
Your list has an arm64 Ventura

**Reply by RealChuan** [reply]
This is the list on the bottles website, not the list obtained using brew fetch.
Now it seems that the bottle version cannot be specified. Is it possible to install the package of bottle version of arm64 Ventura only in macos Ventura?
Or do you have plans to expand this feature, you can set an environment variable to select a bottle version

**Reply by SMillerDev** [reply]
There are no plans to specify bottle versions in any install configuration. The bottles are build specifically for the OS that they specify, so I can't think of any reason to install the wrong bottle on purpose.

**Reply by p-linnane** [reply]
What is your goal here? That would help us understand how to help you. As @SMillerDev said, `brew` will pick the latest available bottle. There should be no need to install a Ventura bottle on Sequoia, so any additional info you can share would be appreciated.

**Reply by RealChuan** [reply]
I develop programs on Sequoia and use brew to manage third-party dependencies. I want my programs to support as many versions of macos as possible, so I want brew to have a choice of different bottles.
For example, libmpv, which relies on a large number of third libraries, and the compilation tool chain is not the same, and vcpkg this source installation package management does not support the third library.

dyld[77417]: Symbol not found: _$ss20__StaticArrayStorageCN
  Referenced from: <1A938DE7-880B-3E3B-A28F-C46AE043A1C1> /Applications/myapp.app/Contents/Frameworks/libmpv.2.dylib (built for macOS 15.0 which is newer than running OS)
  Expected in:     <7DB1818B-1D9A-328E-821B-21A33442746A> /usr/lib/swift/libswiftCore.dylib
zsh: abort      /Applications/myapp.app/Contents/MacOS/Player

**Reply by SMillerDev** [reply]
Yeah, that's not really a supported usage of brew.

Additionally, general compatibility only works one way. So you'd be much better off trying to run a VM or something with the oldest version you want to support, and distributing that to newer OSes.
